cmd/kube-apiserver/app/encryption/encryption.go

```diff
@@ -38,31 +43,50 @@ const (
 	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
 	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
 	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
+	envelopeTransformerPrefixV1  = "k8s:enc:envelope:v1:"
 )
 
-// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file
-func GetTransformerOverrides(filepath string) (map[schema.GroupResource]value.Transformer, error) {
```

Comment:
I expected structure like we use for admission plugin configuration.  We provide a format for providing admission plugin configuration information.  API servers can add their admission plugins (KMS plugins in this case), to a plugin registry associated with the apiserver config, and then their config is passed through to them.  This has worked well for admission where we wanted to add multiple plugins (vault is coming), which need their own styles of config, but wil l provide the same interface in the end.  This looks like a very similar problem to me.